includes/create-theme/theme-locale.php

```diff
@@ -17,12 +17,16 @@ public static function escape_string( $string ) {
 		}
 
 		// Check if the text is already escaped.
-		if ( str_starts_with( $string, '<?php echo' ) ) {
+		if (
+			str_starts_with( $string, '<?php echo' ) ||
+			str_starts_with( $string, '<?php esc_html_e' ) ||
+			str_starts_with( $string, '<?php esc_html' )
+		) {
 			return $string;
 		}
```

Comment:
There are plenty [other escaping functions](https://developer.wordpress.org/apis/security/escaping/#toc_1) that might show up. I'm wondering if we could instead try to catch them all with regex:

```suggestion
if (
    preg_match(
        '/^<\?php\s+(echo|esc_)/',
        $string
    )
) {
    return $string;
}
```

All current escaping functions match that pattern, but we could also just add them all to avoid catching unintended functions:

```php
if (
    preg_match(
        '/^<\?php\s+(echo|esc_html_e|esc_html__|esc_html_x|esc_attr_e|esc_attr__|esc_attr_x|esc_url|esc_js|esc_textarea)/',
        $string
    )
) {
    return $string;
}
```

Reply:
Good point but I think we could simplify it by just checking if the string starts with `<?php`. That will cover all the cases in a simpler way. Implemented it here: https://github.com/WordPress/create-block-theme/pull/665/commits/9b223b79813936882f943f1e2910e636e05d32ce